src/Adapter/Console.php

```diff
@@ -164,16 +164,24 @@ public function __destruct()
     public function setOutputStream($resource)
     {
         $fileOpenError = null;
-        ErrorHandler::start();
-        try {
-            $stream = fopen($resource, 'w');
-        } catch (ValueError $fileOpenError) {
+        if (is_null($resource)) {
             $stream = false;
+        } else {
+            ErrorHandler::start();
+            try {
+                $stream = fopen($resource, 'w');
+            } catch (ValueError $fileOpenError) {
+                $stream = false;
+            }
+            $error = ErrorHandler::stop();
         }
-        $error = ErrorHandler::stop();
 
         if ($stream === false) {
-            $previous = $fileOpenError ?: $error;
+            if (isset($error)) {
```

Comment:
`$error` should always be defined upfront - the `ErrorHandler::start()` and `ErrorHandler::stop()` calls are too nested

Reply:
In general `isset($variableThatMayNotExist)` is not valid code under `laminas/`

Reply:
I avoided the complex behavour by passing an optional argument to ErrorHandler::start()